src/Pact/Gas/Table.hs

```diff
@@ -89,6 +89,7 @@ defaultGasTable =
   ,("contains", 2)
   ,("create-module-guard", 1)
   ,("create-pact-guard", 1)
+  ,("create-principal", 1) -- TODO check this
```

Comment:
I think these costs are fine for k:, m:, r: and p:, but for other keyset and user guards there needs to be an additional cost as they are unbounded.

So in addition to these nominal prices, add a `GasCostConfig` (`GPrincipal`). For costing, propose we send the size of the source bytestring for the hash creation and use `memoryCost` on it.

Reply:
then `TODO` can go :)

Reply:
Sounds good 😄

Reply:
Added